engine/src/main/java/io/camunda/zeebe/engine/api/RecordProcessor.java

```diff
@@ -20,6 +22,21 @@ public interface RecordProcessor {
    */
   void init(RecordProcessorContext recordProcessorContext);
 
+  /**
+   * Returns true if the processor is responsible for processing a record with the given valueType.
+   * If it returns true, then {@link RecordProcessor#process(TypedRecord, ProcessingResultBuilder)}
+   * and {@link RecordProcessor#replay(TypedRecord)} must successfully process or replay the record
+   * with the given valueType. The processor can also choose to skip the record depending on the
+   * intent of the record.
+   *
+   * <p>If it returns false, {@link RecordProcessor#process(TypedRecord, ProcessingResultBuilder)} *
+   * and {@link RecordProcessor#replay(TypedRecord)} with the given valueType will not be called.
+   *
+   * @param valueType valueType of a record
+   * @return true or false
+   */
+  boolean canProcess(ValueType valueType);
```

Comment:
:thought_balloon: Nothing you need to change right now, was just thinking whether we want to rename it to something like `accepts`

Reply:
:+1: I like it.